### PR TITLE
Add WinForms exceptions for .NET 6

### PR DESCRIPTION
**Pull request by gewarren**
Contributes to dotnet/docs#23529 and contributes to dotnet/docs#22021.

[DefaultAction preview](https://review.docs.microsoft.com/en-us/dotnet/api/system.windows.forms.datagridviewtopleftheadercell.datagridviewtopleftheadercellaccessibleobject.defaultaction?view=net-5.0&branch=pr-en-us-5484#exceptions)

[Item preview](https://review.docs.microsoft.com/en-us/dotnet/api/system.windows.forms.treenodecollection.item?view=net-5.0&branch=pr-en-us-5484#System_Windows_Forms_TreeNodeCollection_Item_System_Int32_)

### CODE REVIEW

File: xml/System.Windows.Forms/DataGridViewRowStateChangedEventArgs.xml [hunk 1]
```diff
@@ -82,6 +80,7 @@
   
  ]]></format>
         </remarks>
+        <exception cref="T:System.ArgumentNullException">.NET 6 and later versions: <see paramref="dataGridViewRow" /> is <see langword="null" />.</exception>
```
Comment (RussKie) [inline xml/System.Windows.Forms/DataGridViewRowStateChangedEventArgs.xml:83]: This doesn't appear to render correctly:
image

File: xml/System.Windows.Forms/TreeNodeCollection.xml [hunk 2]
```diff
@@ -1278,6 +1278,7 @@ To assign <xref:System.Windows.Forms.TreeNode> objects to a specific location, o
   
  ]]></format>
         </remarks>
+        <exception cref="T:System.ArgumentNullException">.NET 6 and later versions: <see paramref="index" /> is <see langword="null" />.</exception>
```
Comment (RussKie) [inline xml/System.Windows.Forms/TreeNodeCollection.xml:1281]: ditto here
image

File: xml/System.Windows.Forms/DataGridViewRowStateChangedEventArgs.xml (hunk 1 above)
Comment (gewarren) [reply xml/System.Windows.Forms/DataGridViewRowStateChangedEventArgs.xml]: Thanks - good catch!

**Review by RussKie — APPROVED** [review]

**Comment by RussKie** [comment]
Thank you!

**Pull request merged**